### PR TITLE
Fixes #13880: Added dt=

### LINKED ISSUE #13880: Implement dt= like we have afl=, etc
Grouping them by contiguous blocks of code traced

### PR DESCRIPTION
**Pull request by deepakchethan**

### CODE REVIEW

File: libr/debug/trace.c [hunk 1]
```diff
@@ -119,6 +239,13 @@ R_API void r_debug_trace_list (RDebug *dbg, int mode) {
 			}
 		}
 	}
+	if (flag) {
+		RCore *core = dbg->corebind.core;
+		r_list_sort (info_list, cmpaddr);
```
Comment (radare) [inline libr/debug/trace.c]: You cant use rcore from rdebug

File: libr/debug/trace.c [hunk 2]
```diff
@@ -98,16 +99,135 @@ R_API RDebugTracepoint *r_debug_trace_get (RDebug *dbg, ut64 addr) {
...
+		}
+	}
+}
+
 R_API void r_debug_trace_list (RDebug *dbg, int mode) {
 	int tag = dbg->trace->tag;
 	RListIter *iter;
+	bool flag = false;
+	RList *info_list = r_list_new ();
+	if (!info_list && mode == '=') {
```
Comment (radare) [inline libr/debug/trace.c]: thats probably never going to happen. info_list. better use || instead of &&

File: libr/debug/trace.c [hunk 3]
```diff
@@ -119,6 +239,13 @@ R_API void r_debug_trace_list (RDebug *dbg, int mode) {
 			}
 		}
 	}
+	if (flag) {
+		RCore *core = dbg->corebind.core;
+		r_list_sort (info_list, cmpaddr);
+		r_core_visual_list (core, info_list, core->offset, core->blocksize,
```
Comment (radare) [inline libr/debug/trace.c]: i wouldn't go that picky and i would use blocksize = 1 and core->offset passed as argument, so you dont need to add extra dpeendencies

File: libr/debug/trace.c [hunk 4]
```diff
@@ -98,16 +99,135 @@ R_API RDebugTracepoint *r_debug_trace_get (RDebug *dbg, ut64 addr) {
...
+				ListInfo *info = R_NEW0 (ListInfo);
+				char temp[8];
+				if (!info) {
+					return;
+				}
+				info->pitv = (RInterval) {trace->addr, trace->size};
+				info->vitv = info->pitv;
+				info->perm = -1;
+				snprintf (temp, sizeof (temp), "%d", trace->times);
+				info->name = strdup (temp);
```
Comment (radare) [inline libr/debug/trace.c]: use r_str_newf instead of snprintf+strdup

File: libr/debug/trace.c [hunk 5]
```diff
@@ -98,16 +99,135 @@ R_API RDebugTracepoint *r_debug_trace_get (RDebug *dbg, ut64 addr) {
...
 	RDebugTracepoint *trace;
 	r_list_foreach (dbg->trace->traces, iter, trace) {
 		if (!trace->tag || (tag & trace->tag)) {
 			switch (mode) {
 			case 'q':
 				dbg->cb_printf ("0x%"PFMT64x"\n", trace->addr);
 				break;
+			case '=': {
+				ListInfo *info = R_NEW0 (ListInfo);
+				char temp[8];
```
Comment (radare) [inline libr/debug/trace.c]: overflows. just use r_strnewf and kill this var

File: libr/debug/trace.c (hunk 3 above)
Comment (deepakchethan) [reply libr/debug/trace.c]: > i wouldn't go that picky and i would use blocksize = 1 and core->offset passed as argument, so you dont need to add extra dpeendencies

But you said not to use RCore in RDebug. core->offset is from RCore isn't it ?

File: libr/debug/trace.c (hunk 2 above)
Comment (deepakchethan) [reply libr/debug/trace.c]: The idea is that info_list is only required when the mode is dt=. So I think && is the right op here?

File: libr/debug/trace.c (hunk 4 above)
Comment (deepakchethan) [reply libr/debug/trace.c]: Updated this

File: libr/debug/trace.c (hunk 5 above)
Comment (deepakchethan) [reply libr/debug/trace.c]: Okay, updated.

**Pull request merged**